### DISCUSSION TITLE
[3.x] Optional route parameters

### DISCUSSION THREAD

**Discussion by megasteve19** [livewire/livewire#6122] [7. Bugs (archived)]
### Livewire version

3.0@beta

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I'm not sure is this a bug or a missing feature. We can't make optional route parameters for component.

```php
class Edit extends Component
{
    public ?Post $post = null;

    public function mount(?Post $post = null)
    {
        $this->post = $post;
    }

    public function updated($name, $value)
    {
        // ...
    }

    #[Layout('layouts.admin')]
    public function render()
    {
        return view('livewire.admin.posts.create');
    }
}

// ...

Route::get('/{post?}/edit', Posts\Edit::class)->name('edit');	
```

In above code, it will not work at all. Route will only throw `404`. Maybe it's made this way. If this is the way it works I think this should be included in the documentation.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by megasteve19** [comment]
Realized that my example is very wrong, I will play little more with it.

**Comment by jals-es** [comment]
Hi @megasteve19 have you solved the problem? I'm getting an error when using an optional route parameter ant it's not filled.

**Reply by megasteve19** [reply]
If I remember correctly, I couldn't found a solution so I went with duplicating same component without any parameters

**Comment by jals-es** [comment]
I solve it, removing the optional parameter from de mount() arguments and adding it as class variable. 

```
public $post = null;

    public function mount()
    {
        
    }
```

Try this.
